### PR TITLE
[Merged by Bors] - feat(ENNReal): Scalar multiplication by `NNReal` is finite

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

Co-authored-by: Bhavik Mehta <bhavik.mehta8@gmail.com>

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by b-mehta — APPROVED** [review]
I wrote part of this so I won't merge, but LGTM.

File: Mathlib/Data/NNReal/Basic.lean [hunk 1]
```diff
@@ -155,6 +155,9 @@ protected theorem coe_injective : Injective ((↑) : ℝ≥0 → ℝ) := Subtype
 
 @[simp, norm_cast] lemma coe_one : ((1 : ℝ≥0) : ℝ) = 1 := rfl
 
+@[simp] lemma mk_zero : (⟨0, le_rfl⟩ : ℝ≥0) = 0 := rfl
+@[simp] lemma mk_one : (⟨1, zero_le_one⟩ : ℝ≥0) = 1 := rfl
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/NNReal/Basic.lean:159]: Maybe these should be protected

Comment (YaelDillies) [reply Mathlib/Data/NNReal/Basic.lean:159]: Other similarly named lemmas are not protected, so I don't think so

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**